bx_py_utils/hash_utils.py

```diff
@@ -0,0 +1,55 @@
+import hashlib
+import itertools
+import string
+
+
+class UrlSafeEncoder:
+    """
+    Encode bytes into a URL safe string.
+    Note:
+        Use a URL safe alphabet (see RFC 3986) without umlauts
+
+    Remove umlauts will hopefully avoid to generate valid words ;)
+    """
+    alphabet = (
+        '-._~'  # Allowed additional characters RFC 3986
```

Comment:
Why do we have these characters but not `aeiou`?

Reply:
Because if the umlauts are preset, then to may get something like "sex" or other short words you would never like to have in a url ;)